xbmc/types.go

```diff
@@ -145,17 +145,17 @@ type ListItemInfo struct {
 	Exif        string `json:"exif,omitempty"`
 }
 
-// ListItemArt ...
+// ListItemArt, see https://kodi.wiki/view/Artwork_types
 type ListItemArt struct {
-	Thumbnail         string    `json:"thumb,omitempty"`
-	Poster            string    `json:"poster,omitempty"`
-	TvShowPoster      string    `json:"tvshowposter,omitempty"`
-	Banner            string    `json:"banner,omitempty"`
-	FanArt            string    `json:"fanart,omitempty"`
+	Thumbnail         string    `json:"thumb,omitempty"`        //used for episodes only
+	Poster            string    `json:"poster,omitempty"`       //poster + a clearly visible logo or name of the video
+	TvShowPoster      string    `json:"tvshowposter,omitempty"` //TODO: something very old or skin specific? remove it?
+	Banner            string    `json:"banner,omitempty"`       //contain recognizable characters and typically includes a clearly visible logo or name
```

Comment:
That is not a legacy item.

Here is an example of a skin that uses it - https://github.com/jurialmunkey/skin.arctic.fuse/blob/e29dfddb1e12099ea0f4ebe969c8e85db13caec7/shortcuts/skinvariables-images.xml#L41

So we need to return it :)

Reply:
i have not removed TvShowPoster, i just added TODO. i will delete this TODO.

Reply:
but i do not see `tvshowposter` in the link you provided.
in whole repo - i also do not see it: https://github.com/search?q=repo%3Ajurialmunkey%2Fskin.arctic.fuse%20tvshowposter&type=code

Reply:
but i do see `tvshow.poster`. maybe we need to rename our to `tvshow.poster`?

Reply:
i guess it should be renamed in python part here
`                "tvshowposter": _infoLabels["arttvshowposter"],`
https://github.com/elgatito/plugin.video.elementum/blob/214aab5f32bf2f8c553a37419e9f22a2760d11c0/resources/site-packages/elementum/navigation.py#L234

Reply:
I thought Github was highlighting `Banner` line. That comment was about it. I see `Banner` was removed from Arts.

Reply:
About `TvShowPoster` I do not recall really. It can be an Art used by old Kodi.

Reply:
> i guess it should be renamed in python part here

Hard question. It is not that generated by usual ListItem creators. It is generated by this thing for Kodi Player: https://github.com/elgatito/elementum/blob/master/api/infolabels.go

Reply:
> I meant to rename "tvshowposter": part, not infolabel part. But I still think it is some kind of legacy that we do not need anymore.

I try not to remove something that does not hurt me. People was use Elementum even on Kodi 16, don't see why we should limit it (if we can).

Reply:
@elgatito 
As far as I have tested, banners not removed. They are not used by default since tmdb does not supply banners. When fanart.tv is turned on, the banners are there. But @antonsoroko will explain better. We had a discussion about this. I tested it and it works fine.

Reply:
@elgatito about banner: it was not set in version 0.1.99.
i added it in https://github.com/elgatito/elementum/commit/7134070342b694efbb6e4dbe6915c158242399e8

if short: but then matke-84 found out that explicit setting banner to "wrong" image (backdrop height is big, but banner height should be small) ruin skin's fallback logic.
so it is better not to set banner and let skin compose banner from fanart + logo (which i added here, before that skin used fanart+plain text).

long story with screenshots: https://github.com/elgatito/plugin.video.elementum/issues/945#issuecomment-1978622152

Reply:
@elgatito about "tvshowposter":
i think it was added for some specific skin, **but i am sure it never worked**, because it was added with mistake.

see https://github.com/search?q=tvshow.poster+language%3Axml&type=code

skins uses `tvshow.poster` but elementum set `tvshowposter` - so it never was used by skin.
see https://github.com/elgatito/plugin.video.elementum/blob/214aab5f32bf2f8c553a37419e9f22a2760d11c0/resources/site-packages/elementum/navigation.py#L234

i propose to delete it. 
but if for some reason you would like to keep it - then it should be fixed in `navigation.py`.

Reply:
Let's leave `TvShowPoster` as is (and setters for it), so that legacy Kodi and legacy skins could use that field. The logic with Banner is now clear, so I resolve this conversation.

Reply:
@elgatito but if we really want this TvShowPoster to be actually used - we should fix python code.

Reply:
We output `ListItem` fields to Python, and then python is parsing them and setting arts and other fields, so if you think field should have a different name - it should be either change in `json:...` field name for `TvShowPoster` or ythe python code and parses fields.

Reply:
> ListItem.Art(tvshow.poster)

It is interesting how that function reads Listitem properties without dots or in different case. It can be that `ListItem.Art(tvshow.poster)` is the same as `ListItem.Art(tvshowposter)`.

Reply:
And if that dot is really needed, then there are bunch or additional arts listed:

<html>
<body>
<!--StartFragment-->
ListItem.Art(tvshow.poster) | Returns the tv show poster of the parent container. |  
-- | -- | --
ListItem.Art(tvshow.banner) | Returns the tv show banner of the parent container. |  
ListItem.Art(tvshow.clearlogo) | Returns the tv show clearlogo (if it exists) of the parent container. |  
ListItem.Art(tvshow.landscape) | Returns the tv show landscape (if it exists) of the parent container. |  
ListItem.Art(tvshow.clearart) | Returns the tv show clearart (if it exists) of the parent container. |  
ListItem.Art(season.poster) | Returns the season poster of the currently selected season. (Only available in DialogVideoInfo.xml) |  
ListItem.Art(season.banner) | Returns the season banner of the currently selected season. (Only available in DialogVideoInfo.xml) |  
ListItem.Art(season.fanart) | Returns the fanart image of the currently selected season. (Only available in DialogVideoInfo.xml)

<!--EndFragment-->
</body>
</html>

Reply:
I use tvshow.poster in my skin. Means with a dot. And as far as I know, it's always been that way. I don't know, maybe I'm wrong.

Reply:
https://kodi.wiki/view/Artwork/Accessing_with_skins_and_JSON-RPC 

here it has example with dot. but i guess the only way to say for sure it is to read kodi's code.

anyway, i think we should create new issue for these arts with dot.